src/univariate/discrete/negativebinomial.jl

```diff
@@ -129,3 +129,38 @@ function cf(d::NegativeBinomial, t::Real)
     r, p = params(d)
     return (((1 - p) * cis(t)) / (1 - p * cis(t)))^r
 end
+
+# ChainRules definitions
+
+function ChainRulesCore.rrule(::typeof(logpdf), d::NegativeBinomial, k::Real)
+    # Compute log probability
+    r, p = params(d)
```

Comment:
Is this worth unpacking the params and re-writing the primal computation instead of just calling logpdf.

Reply:
Maybe not. I'll revisit this.

Reply:
Hmm I guess just calling `logpdf` would mean that we do the parameter checks twice.

Reply:
I will keep as-is for now.